src/core/ext/filters/client_channel/lb_policy/subchannel_list.h

```diff
@@ -324,8 +324,8 @@ void SubchannelData<SubchannelListType,
   pending_watcher_ =
       new Watcher(this, subchannel_list()->WeakRef(DEBUG_LOCATION, "Watcher"));
   subchannel_->WatchConnectivityState(
-      std::unique_ptr<SubchannelInterface::ConnectivityStateWatcherInterface>(
-          pending_watcher_));
+      std::unique_ptr<SubchannelInterface::ConnectivityStateWatcherInterface>{
```

Comment:
Why is this change necessary?  `std::unique_ptr<SubchannelInterface::ConnectivityStateWatcherInterface>` is a class type, so parens should be fine here.  This is not a cast, it's constructing a new object.

Reply:
This is to address this weird false positive of clang-tidy. Clang-tidy keeps saying it should change so I did it to workaround it.

Reply:
This seems obviously wrong.

Can we instead leave it as-is and use a NOLINT comment to have clang-tidy leave it alone?